### ISSUE TITLE
DEEP NEAREST CLASS MEAN CLASSIFIERS

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#363]
https://openreview.net/pdf?id=rkPLZ4JPM

**Comment by sweaterr** [comment]
In this paper we introduce DeepNCM, a Nearest Class Mean classification method enhanced to directly learn highly non-linear deep (visual) representations of the data.

To overcome the computational expensive process of recomputing the class means after every update of the representation, we opt for approximating the class means with an online estimate.

Moreover, to allow the class means to follow closely the drifting representation we introduce per epoch mean condensation.

Using online class means with condensation, DeepNCM can train efficiently on large datasets.

Our (preliminary) experimental results indicate that DeepNCM performs on par with SoftMax optimised networks.

**Comment by sweaterr** [comment]
In this paper, we introduce our ongoing work on DeepNCM, which learns directly a highly nonlinear deep representation while enforcing the desired property that images will be mapped closer to their class mean than any other class means.
이 논문에서는 DeepNCM에 대한 지속적인 연구를 소개합니다. DeepNCM은 고도의 비선형 깊이 표현을 직접 배우면서 다른 클래스보다 이미지를 클래스 평균에 더 가깝게 매핑 할 수 있습니다.

DeepNCM can be considered as the extension of the Metric Learning approach for NCM (Mensink et al,, 2013), where a shared visual representation has been learned on top of hand designed fixed features, which generalises well over novel classes and therefore could be successfully used for class incremental learning.
DeepNCM은 NCM (Mensink et al ,, 2013)에 대한 메트릭 학습 접근법의 확장으로 간주 될 수 있습니다. 공유 시각적 표현은 손으로 디자인 한 고정 된 기능을 바탕으로 학습되었으며, 새로운 기능의 클래스를 능가하므로 잘 수행 될 수 있습니다 클래스 증분 학습에 사용됩니다.

**Comment by sweaterr** [comment]
We believe that DeepNCM, to date, has not yet been fully explored because computing class means over a changing representation would be overly expensive.
우리는 변화하는 표현에 대한 계급 계산 방법이 지나치게 비싸기 때문에 현재까지 DeepNCM이 완전히 탐구되지 않았다고 믿습니다.

Here, we circumvent expensive updates by approximating class means with an online estimate.
여기에서는 온라인 견적을 사용하여 클래스 평균을 근사하여 비싼 업데이트를 우회합니다.

This approximates the real mean, while guaranteeing high adaptability to the changing deep representation as new samples are processed.
이는 새로운 샘플이 처리 될 때 변화하는 심층 표현에 대한 높은 적응력을 보장하면서 실제 평균을 근사화합니다.

**Comment by sweaterr** [comment]
Relation to other approaches Most notably, our proposed DeepNCMs relate to:.
다른 접근법과의 관계 우리가 제안한 DeepNCM은 다음과 관련이있다.

**Comment by sweaterr** [comment]
The incremental classifier learning approach of Rebuffi et al, (2017), where NCM classifiers are applied on SoftMax learned visual representations.
Rebuffi et al. (2017)의 증분 분류기 학습 접근법. NCM 분류기가 SoftMax에서 배운 시각적 표현에 적용됩니다.

While NCM classifiers are used, they do use a representation learned using a (rather) standard SoftMax classifiers, on a fixed-size training size to meet memory constraints, and with a distillation loss to overcome catastrophic forgetting.
NCM 분류기가 사용되는 동안, (다소) 표준 SoftMax 분류기를 사용하여 배운 표현을 사용하고, 메모리 제약 조건을 충족시키기 위해 고정 크기의 교육 크기로, 그리고 파국적 인 망각을 극복하기 위해 증류 손실을 사용합니다.

Our proposed DeepNCM would allow to directly learn the visual representation.
우리가 제안한 DeepNCM은 시각적 표현을 직접 배울 수 있습니다.

**Comment by sweaterr** [comment]
The few-shot learning approach of Prototypical Networks (Snell et al,, 2017), where a deep representation is learned based on NCM classifiers, however they consider the scenario of episode-few-shot classification (Vinyals et al,, 2016), where each episode is a classification task using just a handful classes and data points.
Prototypical Networks (Snell et al, 2017)에서 NCM 분류기를 기반으로 심도 깊은 표현을 학습 한 몇 안되는 학습 접근법이지만, 에피소드 (극소수) 분류 시나리오 (Vinyals et al ,, 2016)를 고려하면, 각 에피소드는 소수의 클래스와 데이터 포인트를 사용하는 분류 작업입니다.

Computing exact class mean over all images in an episode is in this scenario trivial, In contrast to DeepNCM, their method would not extend to a larger number of images or classes, e.g, when not all classes are present in a (mini-)batch.
에피소드의 모든 이미지에 대한 정확한 클래스 평균 계산은이 시나리오에서 사소한 것으로 DeepNCM과 달리 그들의 메소드는 많은 수의 이미지 나 클래스로 확장되지 않습니다. 예를 들어 모든 클래스가 (최소) 배치에 존재하지 않을 때 .

**Comment by sweaterr** [comment]
Contributions We propose DeepNCM, a NCM-based method for Deep Learning.
우리는 Deep Learning을위한 NCM 기반의 방법 인 DeepNCM을 제안합니다.

Our key contributions are using an online approximation for the class means, together with a procedure which we coin mean condensation, which aim to summarise the mean of previous epochs.
우리의 핵심적인 공헌은 이전 시대의 평균을 요약하는 것을 목표로하는 평균 응축을 만드는 절차와 함께 계급 평균을위한 온라인 근사를 사용하는 것입니다.

We present some exciting results, which show that DeepNCM performs on par with its (standard) SoftMax counterpart.
DeepNCM이 (표준) SoftMax와 동등한 성능을 보여 주는 흥미 진진한 결과를 제시합니다.

We are currently working to extend the DeepNCM method to class Incremental Learning and Open-Set Classification, where it is expected that the proposed NCM-based learned representation will have an edge over SoftMax-based representations.
우리는 현재 DeepNCM 방법을 Incremental Learning 및 Open-Set Classification으로 확장하기 위해 노력하고 있습니다. 제안 된 NCM 기반 학습 표현은 SoftMax 기반 표현보다 우위를 차지할 것으로 예상됩니다.

**Comment by sweaterr** [comment]
## 2 DEEP NEAREST MEAN CLASSIFICATION

The nearest class mean (NCM) classifier is a distance-based classifier, which assigns an image to the class with the closest mean:.
가장 가까운 클래스 평균 (NCM) 분류자는 거리 기반 분류이며 가장 가까운 평균을 갖는 클래스에 이미지를 할당합니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39303139-0c6b57a4-4990-11e8-951e-58ea61ce617b.png)

**Comment by sweaterr** [comment]
where Y denotes the number of classes, class mean µy =
1
Ny
P
i:yi=y xi
, and Ny the number of
examples in class y. The success of NCM classifiers critically depends on the distance metric used.
Mensink et al. (2013) proposed a probabilistic interpretation of the class mean distances, to learn a
(squared) low-rank Mahalanobis distance, by maximising the log-likelihood:

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39303509-66354db6-4991-11e8-884c-d393f856f648.png)

**Comment by sweaterr** [comment]
e. The final
NCM classifier can be identified as a multi-class SoftMax classifier, with constrained bias terms and
weight vectors, e.g. the weight vector is constraint to: wy = W>Wµy, see (Mensink et al., 2013).

**Comment by sweaterr** [comment]
NCM for Learned Representations Instead of relying on a fixed representation and learning a Mahalanobis metric W, the current standard is to learn the parameters of a deep representation φ(·).
학습 표현에 대한 NCM 고정 표현에 의존하고 마할 라 노비스 (Mahalanobis) 미터법 W를 학습하는 대신 현재의 표준은 깊은 표현 φ (·)의 매개 변수를 학습하는 것입니다.

We postulate that the highly non-linear nature of deep representations eliminates the need of a linear metric W and allows to use the Euclidean distance between the deep representations:.
우리는 딥 리 프리젠 테이션의 비선형 성은 선형 메트릭 W가 필요 없으며 깊은 표현 사이에서 유클리드 거리를 사용할 수 있다고 가정합니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39305019-a196be58-4996-11e8-80b1-0a93223a27b6.png)

**Comment by sweaterr** [comment]
For DeepNCM, we therefore propose to use an online estimate of the class means:.
DeepNCM의 경우, 따라서 우리는 클래스 수단의 온라인 견적을 사용하도록 제안합니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39305861-81ee2df4-4999-11e8-9006-c5f24987e31a.png)

**Comment by sweaterr** [comment]
where nyi denotes the current number of samples in class yi.
여기서 nyi는 클래스 yi의 현재 샘플 수를 나타냅니다.

The class means are updated with the relevant samples every batch, following smoothly the drifting learned representation φ(·).
클래스 수단은 표류하는 배움 표현 φ (·)를 매끄럽게 따르는 모든 배치에서 관련 샘플로 업데이트됩니다.

**Comment by sweaterr** [comment]
### Mean Condenstation 

When computing the online mean, the most recent example counts as much
as the first example, while the representation has drifted since. To assign more weight to recent
examples, we introduce per epoch mean condensation. At the start of each epoch, we set ny =
1 ∀y∈Y , i.e. the historical class means count just as a single data point in the current epoch. Based
on preliminary results, we conclude that mean condensation is beneficial for training, see Fig. 1.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39307394-a764d8ea-499d-11e8-83fc-f766279cd199.png)

**Comment by sweaterr** [comment]
## 3 EXPERIMENTS

In this section we describe some of our initial experiments using the proposed DeepNCM classifier on the MNIST, CIFAR-10 and CIFAR-100 datasets, reporting Top-1 accuracy.
이 절에서는 MNIST, CIFAR-10 및 CIFAR-100 데이터 세트에서 제안 된 DeepNCM 분류기를 사용하여 Top-1 정확도를보고하는 초기 실험 중 일부를 설명합니다.

All models are trained using SGD with Nesterov momentum, with dropout and explicit weight regularisation.
모든 모델은 네 스테 로프 모멘텀이있는 SGD와 드롭 아웃 및 명시 적 무게 정규화를 사용하여 교육됩니다.

**Comment by sweaterr** [comment]
## Comparison to SoftMax 

First, we compare the NCM classifiers to SoftMax classifiers, the current
standard for image classification, on multi-class image classification. We evaluate a (fixed) pixel
based representation, a shallow ConvNet, and a ResNet-32He et al. (2015) architecture. From the
ResNet-32 network point of view, the NCM classifiers uses the penultimate layer as φ(x) and the
class means to learn the network from scratch. The results in Tab. 1a show that DeepNCM is able to
learn strong visual representations, and performs on par with the SoftMax classifier.

**Comment by sweaterr** [comment]
## Generalisation to new classes

A selling point of the NCM over fixed representations is that it generalises well to new classes.
고정 된 표현에 비해 NCM의 판매 시점은 새로운 클래스에 대해 일반적으로 잘 나타납니다.

Here we perform a preliminary generalisation experiment using the CIFAR-100 dataset, split in two subsets, see Tab 1b.
여기서는 CIFAR-100 데이터 세트를 사용하여 두 개의 하위 집합으로 분할하여 예비 일반화 실험을 수행합니다 (표 1b 참조).

Starting from the learned representation on the C50 subset, we either evaluate NCM using the means of the novel classes or we do fine-tuning for 1 (or 5) epochs on the G50 subset.
C50 하위 집합의 학습 된 표현에서 시작하여 우리는 새로운 클래스의 수단을 사용하여 NCM을 평가하거나 G50 하위 집합에서 1 (또는 5) 에포크를 미세 조정합니다.

In the latter case, we either use the SoftMax classifier or the DeepNCM classifier on online means.
후자의 경우 온라인 방법으로 SoftMax 분류 자 ​​또는 DeepNCM 분류자를 사용합니다.

The results indicate that DeepNCM and SoftMax both learn representations which generalise well and comparably to novel classes.
결과는 DeepNCM과 SoftMax가 모두 일반화 된 표현과 새로운 클래스와 비교할 수있는 표현을 학습한다는 것을 나타냅니다.
